### PR TITLE
Add new constructors with ModelReader parameter

### LINKED ISSUE #658: Using a ModelReader with a TranslatorPool
It seems that a `ModelReader` can only be used directly with the `Translator` class through `set_model` and not with a `TranslatorPool`. Is there a workaround for this? If not, would you be interested if I worked for a PR to add this functionality?

### PR DESCRIPTION
**Pull request by panosk**
Added new constructors both in `TranslatorPool` and `Translator` classes that allow easy integration of custom `ModelReader`s. Closes #658.

### CODE REVIEW

File: src/translator.cc [hunk 1]
```diff
@@ -56,7 +56,15 @@ namespace ctranslate2 {
                          Device device,
                          int device_index,
                          ComputeType compute_type) {
-    set_model(models::Model::load(model_dir, device, device_index, compute_type));
+      set_model(models::Model::load(model_dir, device, device_index, compute_type));
```
Comment (guillaumekln) [inline src/translator.cc:59]:
```suggestion
    set_model(models::Model::load(model_dir, device, device_index, compute_type));
```

File: src/translator.cc [hunk 2]
```diff
@@ -56,7 +56,15 @@ namespace ctranslate2 {
...
+  }
+
+  Translator::Translator(models::ModelReader& model_reader,
+                         Device device,
+                         int device_index,
+                         ComputeType compute_type)
+  {
+      set_model(models::Model::load(model_reader, device, device_index, compute_type));
```
Comment (guillaumekln) [inline src/translator.cc:67]:
```suggestion
    set_model(models::Model::load(model_reader, device, device_index, compute_type));
```

File: src/models/model.cc [hunk 3]
```diff
@@ -546,7 +546,7 @@ namespace ctranslate2 {
     }
 
     std::vector<std::shared_ptr<const Model>>
-    load_replicas(const std::string& model_path,
+    load_replicas(models::ModelReader& model_reader,
```
Comment (guillaumekln) [inline src/models/model.cc:549]: As the error in the CI indicates, this function is also used in the Python wrapper. I suggest adding an overload taking the `std::string`, which calls the function taking the model reader.

**Review by guillaumekln — CHANGES_REQUESTED** [review]

**Review by guillaumekln — APPROVED** [review]
Thanks!

**Comment by panosk** [comment]
Thanks for accepting this PR! One final commit to fix for some minor formatting inconsistencies.

**Pull request merged**